### PR TITLE
Casmuser 2906 main

### PR DESCRIPTION
**Pull request by keopp-hpe**
## Summary and Scope

Support for BiCAN.  

## Issues and Related PRs

* Resolves CASMUSER-2906]

### Tested on:

  * `hela`

### Test description:

Installed on hela.  Tested various combinations of CAN, CHN, and DIRECT user access configurations.  Also tested CAN on another system that didn't support CHN.

- Were the install/upgrade-based validation checks/tests run (goss tests/install-validation doc)? N/A
- Were continuous integration tests run? If not, why? N/A
- Was upgrade tested? If not, why? N/A
- Was downgrade tested? If not, why? N/A
- Were new tests (or test issues/Jiras) created for this change? N/A

## Risks and Mitigations

Low risk

## Pull Request Checklist

- [X] Version number(s) incremented, if applicable
- [X] Copyrights updated
- [X] License file intact
- [X] Target branch correct
- [X] CHANGELOG.md updated
- [X] Testing is appropriate and complete, if applicable
- [N/A] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: docs/portal/developer-portal/operations/Configure_Interfaces_on_UANs.md [hunk 1]
```diff
@@ -5,11 +5,13 @@ Perform this procedure to set network interfaces on UANs by editing a configurat
...
 
-If the HPE Cray EX CAN is required, set `uan_can_setup : yes` in `host_vars/XNAME/customer_net.yml` for each node that will use the CAN, or in `group_vars/all/customer_net.yml` to enable the HPE Cray EX CAN on all UANs and Application nodes. The YAML files in `group_vars/all` define settings for every node in the system, whereas the YAML files in `host_vars/XNAME` define settings only for an individual node. Administrators can use either directory or both. When both YAML directories are used, the settings in `host_vars/XNAME` override the settings in `group_vars/all` files. 
+Admins must create the `customer_net.yml` file and use the variables described in this procedure to define the interfaces and routes.
+
+If the HPE Cray EX CAN or CHN is required, set `uan_user_access_cfg` to `CAN` or `CHN` in `customer_net.yml`, depending on whether the CAN or CHN user access network is desired.
```
[the remark this answers is not in the document]
Comment (alanm-hpe) [reply docs/portal/developer-portal/operations/Configure_Interfaces_on_UANs.md:14]: Should we also mention setting `uan_user_access_cfg` to `DIRECT` somewhere in this guide?

File: ChangeLog.md [hunk 2]
```diff
@@ -5,7 +5,16 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
 ## [Unreleased]
+
+## [2.3.1] - 2022-02-15
+- Fix uan_packages to correctly check signatures
+- Updates CFS roles to ROCM 4.5.2
+- Updates CFS roles to AMD 21.40.2
+- Updates CFS roles to Nvidia SDK 21.9
+- Extra testing added to CFS plays
 - Specify Application as a target group in CFS documentation
+## [2.3.1] - 2022-02-16
```
[the remark this answers is not in the document]
Comment (keopp-hpe) [reply ChangeLog.md]: Grabbed 2.3.2 as the next in line.

File: docs/portal/developer-portal/upgrade/Notable_Changes.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+* UAN 2.2 was an internal release and was not made generally available.
+
+#### UAN 2.3.1
+
+* UAN 2.3 no longer ships a default recipe or image. To build a UAN image, administrators should select a COS recipe to build as the base of their UAN and Application Nodes.
+* The role `uan_packages` will now install the rpms needed by UAN and Application Nodes
+* The role `uan_packages` role supports GPG checking when the CSM version is 1.2 or greater.
+  * `uan_disable_gpg_check: yes` must be set if CSM is earlier than 1.2
+  * `uan_disable_gpg_check: no` should be set if CSM is 1.2 or greater
+
```
[the remark this answers is not in the document]
Comment (keopp-hpe) [reply docs/portal/developer-portal/upgrade/Notable_Changes.md:18]: Added a section on BiCAN and the ability to set a default route other than CAN/CHN.

File: docs/portal/developer-portal/operations/Configure_Interfaces_on_UANs.md [hunk 4]
```diff
@@ -5,11 +5,13 @@ Perform this procedure to set network interfaces on UANs by editing a configurat
...
 
-Admins must create the `host_vars/XNAME/customer_net.yml` file and use the variables described in this procedure to define the interfaces and routes.
+Network configuration settings are defined in a `customer_net.yml` file which is used by the Configuration Framework Service \CFS\).  The path to the `customer_net.yml` file in the `uan-config-management` VCS repo will be `group_vars/NODE_GROUP/customer_net.yml` for settings common to all nodes in a given `NODE_GROUP`. `NODE_GROUP` should be replaced by the role and subrole defined in HSM for the nodes - such as `Application_UAN` if the nodes role is `Application` and subrole is `UAN`. Network configuration settings may be defined per node in `host_vars/XNAME/customer.yml`, where `XNAME` is the xname of the node.  These settings would override any settings in the `group_vars/NODE_GROUP/customer.yml` for the node with the given xname.
```
[the remark this answers is not in the document]
Comment (keopp-hpe) [reply docs/portal/developer-portal/operations/Configure_Interfaces_on_UANs.md]: Removed the customer_net.yml and worded it more flexibly.

File: docs/portal/developer-portal/operations/Configure_Interfaces_on_UANs.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (keopp-hpe) [reply docs/portal/developer-portal/operations/Configure_Interfaces_on_UANs.md]: Mentioned the default setting.

File: docs/portal/developer-portal/operations/Create_UAN_Boot_Images.md [hunk 5]
```diff
@@ -131,31 +131,6 @@ Replace `PRODUCT_VERSION` and `CRAY_EX_HOSTNAME` in the example commands in this
      create mode 100644 group_vars/Application_UAN/vars.yml
     ```
```
[the remark this answers is not in the document]
Comment (keopp-hpe) [reply docs/portal/developer-portal/operations/Create_UAN_Boot_Images.md:133]: Put in a reference to the procedure and updated the references in About_UAN_Configuration.

[the remark this answers is not in the document]
Comment (alanm-hpe) [reply docs/portal/developer-portal/operations/Create_UAN_Boot_Images.md:133]: I still think an separate step that explicitly calls out CAN/CHN/DIRECT is needed here (in addition to the link to the doc). If CAN or CHN is not selected, and no custom customer routes are set, the UAN will boot without a default route. By having an individual step, the admins must make a choice of what to use (CAN/CHN/DIRECT).

[the remark this answers is not in the document]
Comment (keopp-hpe) [reply docs/portal/developer-portal/operations/Create_UAN_Boot_Images.md:141]: Added a bold note on selecting the value for user access configuration.

**Review by alanm-hpe — APPROVED** [review]

**Pull request merged**